### DISCUSSION TITLE
how add a new collection to an STAC catalog eoAPI?

### DISCUSSION THREAD

**Discussion by automataIA** [developmentseed/eoAPI#188] [Q&A]
I'm using python. How to add a new collection(empty collection) to an existing STAC catalog created with docker compose custom eoAPI, via python?
 And then, how do you add the items(json, dict) to the collection?  in the 'cleanest' way possible?

**Comment by zacdezgeo** [comment]
Hi @automataIA! The answer depends on a few things, but have you checked out the examples in https://github.com/vincentsarago/MAXAR_opendata_to_pgstac to get started? The best practice is using `pystac` to build your catalog with something like this [example for Maxar](https://github.com/vincentsarago/MAXAR_opendata_to_pgstac/blob/main/Maxar/generate.py). Then, you can use `pypgstac load collections collections.json --dsn postgresql://username:password@0.0.0.0:5439/postgis --method insert_ignore`. 

Would you like to add new items to the collections as new assets becomes available?

**Reply by automataIA** [reply]
> Would you like to add new items to the collections as new assets becomes available?

@zacharyDez yes. I am calculating spectral index(some every day) and then I want to add them to the existing collection.
I'm using the command(they are json of items that have been built and modified starting from items sentinel 2):
```
from pypgstac.db import PgstacDB
from pypgstac.load import Loader
loader.load_items([json_ndvi])
```

How does that sound to you?  If you have more practical and faster solutions, please let me know.

**Reply by XinboEddie** [reply]
Hey @zacharyDez , I also have a similar question. Say we have a collection.ndjson and items.ndjson that are loaded to postgres using the `pypgstac load --method insert-ignore` commands, and we have a weekly data refresh frequency (like new images coming in every week), what would be a good way to perform this upsert step?

**Reply by vincentsarago** [reply]
maybe `pypgstac load ... --method upsert`

**Reply by XinboEddie** [reply]
Is there an alternative API method? @vincentsarago

**Reply by vincentsarago** [reply]
@XinboEddie  depends if you allowed the Transaction Extension to you stac endpoints

**Reply by XinboEddie** [reply]
Is there a documentation for enabling the Transaction extension? I was looking at the repos and didn't find any... this is exactly what I'm looking for and thanks for mentioning this! @vincentsarago

**Reply by zacdezgeo** [reply]
Hi Xinbo!  Here's a link to the collection transaction extension: https://github.com/stac-api-extensions/collection-transaction, and the transaction extension (items): https://github.com/stac-api-extensions/transaction

**Reply by vincentsarago** [reply]
@XinboEddie by default no extensions are enabled

you can see how we customize our devseed eoapi application to enable/disable extensions 

https://github.com/developmentseed/eoapi-devseed/blob/main/runtimes/eoapi/stac/eoapi/stac/app.py#L45C1-L77

but if you just need to enable the transaction extension you'll do something like 

```python
api = StacApi(
    extensions=[
        TransactionExtension(
            client=TransactionsClient(),
            settings=settings,
            response_class=ORJSONResponse,
        )
    ],
    ...
)
```
